File: src/TodoApi/Models/TodoItem.cs
using System;
using Microsoft.EntityFrameworkCore;

namespace Todo.Models {
    public class TodoItem {
        public long Id { get; set; }
        public string Description { get; set; } = string.Empty;
        public DateTime DateCreated { get; set; }
        public DateTime? DateTarget { get; set; }
        public DateTime? DateCompleted { get; set; }
        public TodoState State { get; set; }
    }
}

File: src/todoweb/src/app/models/todo-item.ts
import { TodoState } from "./todo-state";

export class TodoItem {
    id: number;
    description: string;
    dateCreated: Date;
    dateTarget: Date;
    dateCompleted?: Date | undefined;
    state: TodoState;

    constructor(
        id: number,
        description: string,
        dateTarget: Date) {
        this.id = id;
        this.description = description;
        this.dateCreated = new Date();
        this.dateTarget = dateTarget;
        this.state = TodoState.InProgress;
    }
}